src/x11_utils.rs

```diff
@@ -37,6 +37,47 @@ pub trait TryParse: Sized {
     fn try_parse(value: &[u8]) -> Result<(Self, &[u8]), ParseError>;
 }
 
+/// A representation of the header of a request.
+#[derive(Debug, Clone, Copy)]
+pub struct RequestHeader {
+    /// The major opcode of the request.
+    pub major_opcode: u8,
+    /// The minor opcode of the request (which, for some requests, may not be an
+    /// opcode at all).
```

Comment:
Now that I read this, you seem to be aware of this... So I guess I am wrong about `ChangeSaveSet` being broken.

Reply:
`CreateWindowRequest` has a similar thing and it does have a test.